src/compiler/sys/stencil-sys.ts

```diff
@@ -505,7 +505,7 @@ export const createSystem = (c?: { logger?: Logger }) => {
     return results;
   };
 
-  const fetch = global.fetch;
+  const fetch = self?.fetch ?? global?.fetch ?? window?.fetch;
```

Comment:
Truthfully, this can be executed in multiple spaces. So what I'm trying to do here is worker, then node, then window. I'm open to swapping the order here too.

Reply:
I couple questions/maybe asks here -

- How do `self` and `window` differ here?
- Can we add a comment here to explain how/why we chose this order?
- In what circumstances do we use the `fetch` on `self`, `global`, & `window`?

Reply:
`self` is the global namespace object used within a [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers#web_workers_api).

`window` is the browser's global namespace object (I reorganized this to check the reference on that second)

`global` is Node's global namespace object. https://nodejs.org/api/globals.html#globals_global

I did switch these to do worker => browser => node

Reply:
👍 I think that's what I'm looking for in the the ask for a comment. ATM, the comment is kind of repeating the code:
```typescript
  // Reference fetch from a worker thread, then a browser thread, then a node thread.
  const fetch =
    typeof self !== 'undefined'
      ? self?.fetch
      : typeof window !== 'undefined'
      ? window?.fetch
      : typeof global !== 'undefined'
      ? global?.fetch
      : undefined;
```

which doesn't do "future Will and Ryan" much good. Ideally we explain why we choose this order (why is window over global, and self over both of them?) and potentially the context they're used in (what you've got in the [previous comment](https://github.com/ionic-team/stencil/pull/3012#discussion_r693105559))

Reply:
Added!